src/main.rs

```diff
@@ -33,41 +42,47 @@ const PET_OFF: Expiration = OneShot(TimeSpec::new(0, 900_000_000));
 // set line low on death
 
 struct Petter {
-    hand: Lines<Output>,
+    lane: u32,
+    hand: Request,
     timer: TimerFd,
-    values: Cycle<IntoIter<(bool, Expiration), 2>>,
+    values: Cycle<IntoIter<(Value, Expiration), 2>>,
 }
 
 impl Petter {
     fn new() -> Result<Self> {
-        let pin = 25;
-        let chip = Chip::new("gpiochip2").context("Failed to get GPIO chip")?;
-        //FIXME: Since the consumer is set instead of name, this clears on program exit. Once names
-        //are updated, this check should be reinstated.
-        //let label = "PET_WDT";
-        //let consumer = chip.line_info(pin)?.consumer;
-        //ensure!(label == consumer, "Invalid GPIO Pin label, expected {:?}, found {:?}", label, consumer);
-        let opts = Options::output([pin]).values([false]);
-        let line = chip.request_lines(opts).context("Failed to get GPIO pin")?;
+        
+        let line = gpiocdev::find_named_line(GPIO_LABEL)
```

Comment:
The issue with `find_named_line()` (and similar functions in other libraries) is that there's no kernel interface for asking for a line by name, it only understands chip/number. The implementation of these kinds of functions instead iterate through every GPIO line on the system, asking each one what their name is, until it finds a match.

Opening by chip + number and then verifying the name instead goes directly to the line that we want. This has the added benefit that it double checks this program and the device tree agree on what line the watchdog is on. This program could have picked the wrong name, or the device tree could have assigned the right name to the wrong pin. For something as safety critical as the watchdog system it's good to be thorough.